pkg/destroy/azure/azure.go

```diff
@@ -186,13 +200,32 @@ func (o *ClusterUninstaller) Run() (*types.ClusterQuota, error) {
 		waitCtx,
 		func(ctx context.Context) {
 			o.Logger.Debugf("deleting application registrations")
-			err = deleteApplicationRegistrations(ctx, o.applicationsClient, o.serviceprincipalsClient, o.Logger, o.InfraID)
+			// Azure Stack cloud cannot run in Mint mode
+			if o.CloudName == azure.StackCloud {
+				cancel()
+				return
+			}
+			o.Logger.Debugf("Using new Graph API")
+			err = deleteAppRegistrationsGraphAPI(o.graphClient, o.Logger, o.InfraID)
 			if err != nil {
-				o.Logger.Debug(err)
-				if isAuthError(err) {
-					cancel()
-					errs = append(errs, errors.Wrap(err, "unable to authenticate when deleting application registrations and their service principals"))
+				if oDataError, ok := err.(*odataerrors.ODataError); ok {
+					dataErr := oDataError.GetError()
+					if strings.Contains(*dataErr.GetCode(), "RequestDenied") {
+						o.Logger.Debug("Re-trying with legacy API")
+						err = deleteApplicationRegistrations(ctx, o.applicationsClient, o.serviceprincipalsClient, o.Logger, o.InfraID)
+						if err != nil {
+							o.Logger.Debug(err)
+							if isAuthError(err) {
+								cancel()
+								errs = append(errs, errors.Wrap(err, "unable to authenticate when deleting application registrations and their service principals"))
+							}
+							return
+						}
+						cancel()
+						return
+					}
 				}
+				o.Logger.Debug(err)
```

Comment:
`err` should not be logged, but should be appended to the `errs` slice. Perhaps you need to extract the error details and wrap that message.